### DISCUSSION TITLE
Extensions should allow adding buttons to the Composer UI for new features

### DISCUSSION THREAD

**Discussion by goergenj** [microsoft/BotFramework-Composer#3318] [General]
UI-Plug-Ins should allow adding buttons to the Composer UI for new features.

Currently we need to have additions for:
- Logout of users logged in via AAD Authentication plug-in.
- Lock / Unlock of Bots (required as part of the multi-user setup of a hosted Composer)
- Hide the start bot button
- replace start bot button with our own action / test link

@joaollq @sgellock @benbrown

**Comment by cwhitten** [comment]
@a-b-r-o-w-n to consider these scenarios are part of UI plugin work

**Comment by goergenj** [comment]
@cwhitten Any update on potential timeline for this?

**Comment by a-b-r-o-w-n** [comment]
@goergenj we are investing in the extension platform for this release cycle (11/11). We can consider this during that work, but may want to defer to the next release as we develop the core capabilities. 

If you have thoughts or want to propose a design for this, that would be welcomed!

**Comment by goergenj** [comment]
@benbrown @cwhitten Any update on potential timeline for this?

**Comment by goergenj** [comment]
@cwhitten Any update on potential timeline for this?

**Comment by cwhitten** [comment]
@a-b-r-o-w-n can you update @goergenj on where we are at with this?

**Comment by a-b-r-o-w-n** [comment]
@goergenj This is a feature I would like to consider for the first half of 2021. We have developed a simple extension ecosystem that only exposes a few UI contribution points (check out https://github.com/microsoft/BotFramework-Composer/blob/main/extensions/sample-ui-plugin/README.md for an example). I see multiple contribution points in this feature request that I would like to discuss further to get more concrete use cases.

> Logout of users logged in via AAD Authentication plug-in.
Lock / Unlock of Bots (required as part of the multi-user setup of a hosted Composer)

Is there a particular place in the UI that these should be embedded? 

> Hide the start bot button

Does this need to happen conditionally? What data is needed to determine this?

> replace start bot button with our own action / test link

The newest version of Composer has changed the UI for starting bots to support a multi-bot scenario. What is the use-case you have for needing this? Maybe we can incorporate that into something that can be configured through extensions rather than replacing the UI.
